packages/cadl-ranch-specs/http/azure/core/main.cadl

```diff
@@ -49,3 +58,101 @@ Expected response body:
 ```
 """)
 op createOrUpdate is ResourceOperations.ResourceCreateOrUpdate<User>;
+
+@scenario
+@doc("Creates or repalces a User")
+@summary("Adds a user or repalces a user's fields.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected path parameter: id=1
+
+Expected input body:
+```json
+{
+  "name": "Madge"
+}
+```
+
+Expected response body:
+```json
+{
+  "id": 1,
+  "name": "Madge"
+}
+```
+""")
+op createOrReplace is ResourceOperations.ResourceCreateOrReplace<User>;
+
+@scenario
+@doc("Gets a User")
+@summary("Gets a user.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected path parameter: id=1
+
+Expected response body:
+```json
+{
+  "id": 1,
+  "name": "Madge"
+}
+```
+""")
+op get is ResourceOperations.ResourceRead<User>;
+
+@scenario
+@doc("Lists all Users")
+@summary("Lists all users.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Should not generate visible model like CustomPage.
+
+Expected response body:
+```json
+{
+  "value":[
+     {
+        "id":1,
+        "name":"Madge"
+     },
+     {
+        "id":2,
+        "name":"John"
+     }
+  ]
+}
+```
+""")
+op list is ResourceOperations.ResourceList<User>;
+
+@scenario
+@doc("Deletes a User")
+@summary("Deletes a user.")
+@scenarioDoc("""
+Expected path parameter: id=1
+
+Expected response of status code 204 with empty body.
+""")
+op delete is ResourceOperations.ResourceDelete<User>;
+
+@scenario
+@doc("Exports a User")
+@summary("Exports a user.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected path parameter: id=1
+Expected query parameter: format=json
+
+Expected response body:
+```json
+{
+  "id": 1,
+  "name": "Madge"
+}
+```
+""")
+op export is ResourceOperations.ResourceAction<User, UserExportParams, User>;
```

Comment:
The path is `/azure/core/users/{:id}:export`, which is a bit weird to me, as I assume we should be able to use `/azure/core/users:export?...` as well